File: niuniu-take-out-springboot3/server/src/main/java/common/niuniu/service/EmployeeService.java
package common.niuniu.service;

import common.niuniu.dto.EmployeeDTO;
import common.niuniu.dto.EmployeeFixPwdDTO;
import common.niuniu.dto.EmployeeLoginDTO;
import common.niuniu.dto.PageDTO;
import common.niuniu.entity.Employee;
import common.niuniu.result.PageResult;

public interface EmployeeService {
    Employee getEmployeeById(Integer id);

    Employee login(EmployeeLoginDTO employeeLoginDTO);

    void register(EmployeeLoginDTO employeeLoginDTO);

    PageResult employeePageList(PageDTO pageDTO);

    void update(EmployeeDTO employeeDTO);

    void delete(Integer id);

    void onOff(Integer id);

    void addEmployee(EmployeeDTO employeeDTO);

    void fixPwd(EmployeeFixPwdDTO employeeFixPwdDTO);
}

File: niuniu-take-out-kotlin/server/src/main/kotlin/common/niuniu/service/EmployeeService.kt
package common.niuniu.service

import common.niuniu.dto.EmployeeDTO
import common.niuniu.dto.EmployeeFixPwdDTO
import common.niuniu.dto.EmployeeLoginDTO
import common.niuniu.dto.PageDTO
import common.niuniu.po.Employee
import common.niuniu.result.PageResult

interface EmployeeService {
    fun getEmployeeById(id: Int?): Employee?

    fun login(employeeLoginDTO: EmployeeLoginDTO?): Employee?

    fun register(employeeLoginDTO: EmployeeLoginDTO?)

    fun employeePageList(pageDTO: PageDTO?): PageResult?

    fun update(employeeDTO: EmployeeDTO?)

    fun delete(id: Int?)

    fun onOff(id: Int?)

    fun addEmployee(employeeDTO: EmployeeDTO?)

    fun fixPwd(employeeFixPwdDTO: EmployeeFixPwdDTO?)
}
